Sources/MultiSets/CountedSet/CountedSet.swift

```diff
@@ -0,0 +1,41 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift Collections open source project
+//
+// Copyright (c) 2021 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+//
+//===----------------------------------------------------------------------===//
+
+/// An unordered, counted multiset.
+@frozen
+public struct CountedSet<Element: Hashable>: RawRepresentable {
+  // Allows internal setter to be referenced from inlined code
+  @usableFromInline
+  internal var _storage = RawValue()
+
+  @inlinable @inline(__always)
+  public var rawValue: [Element: UInt] { _storage }
+
+  @inlinable
+  public var isEmpty: Bool { rawValue.isEmpty }
+
+  /// Creates an empty counted set with preallocated space for at least the
+  /// specified number of unique elements.
+  ///
+  /// - Parameter minimumCapacity: The minimum number of elements that the
+  ///   newly created counted set should be able to store without reallocating
+  ///   its storage buffer.
+  @inlinable
+  public init(minimumCapacity: Int) {
+    self._storage = .init(minimumCapacity: minimumCapacity)
+  }
+
+  @inlinable
+  public init?(rawValue: [Element: UInt]) {
```

Comment:
This is the initializer required by `RawRepresentable`: filtering out items would mean that the input wasn’t used as a `rawValue`, which would violate the protocol.

I don’t intend for this to be used often; it’s mainly important for deserialization and similar tasks.

Reply:
I meant to say I think we should get rid of the `RawRepresentable` conformance! (It comes with interoperability overtones and we don't need any of the functionality it provides.)

Reply:
I disagree about removing `RawRepresentable`, but I’ll acquiesce.

In that case, do we even need a dictionary initializer? I definitely don’t think a nonfailable unlabeled initializer is appropriate, as it would not be value preserving.

Reply:
I know I said I’d acquiesce, but upon reflection a lot of the desired functionality _is_ replicating `RawRepresentable`, particularly with regards to serialization and equality-checking. If I dropped `RawRepresentable`, I’d just have to replicate its implementations explicitly.

Moreover, this type has a similar role to `OptionSet`: most operations are actually calling a different operation on the raw value, and it implements set operations backed by a non-set type. I’d argue that’s a significant precedent.

`RawRepresentable` is an extremely simple protocol, and literally every part of it is relevant. Moreover, guaranteeing lossless conversion to and from a dictionary makes it much easier for users to implement exotic forms of serialization for.

You keep saying it isn’t needed, but I don’t see how that justifies its exclusion. Is there some undesired effect of conformance that I’m unaware of?

Note that `CountedMultiset` could expose the underlying value as both `rawValue` _and_ `elements`, and add other initializers. For instance, I think there should be an unsafe variant of initializer that performs no checks on the dictionary whatsoever as a performance optimization.